### PR TITLE
Fix Auth0 Production Cron Job Check of Inactive Users 

### PR DESCRIPTION
**Pull request by vahidbazzaz**
The PR switches the PHP formatter to `junstyle.php-cs-fixer` due to the discontinuation of the previous one, and updates the query to check only active bases and fix the cron job for inactive users.

### CODE REVIEW

**Comment by vahidbazzaz** [comment]
[BUG dropapp/auth0 production cron job check of inactive users fails due to new base removal script](https://trello.com/c/68xRRHny/1460-bug-dropapp-auth0-production-cron-job-check-of-inactive-users-fails-due-to-new-base-removal-script)

File: library/lib/auth0.php [hunk 1]
```diff
@@ -246,14 +255,17 @@ function isUserInSyncWithAuth0($userId)
             $validationResult['valid_lastday'] = (!empty($auth0User['app_metadata']['valid_lastday']) && $auth0User['app_metadata']['valid_lastday'] == $dbUser['valid_lastday']) ? 'true' : 'false';
         }
 
+        $false_key = array_search('false', $validationResult);
+        $return_value = !$false_key;
+    } elseif ($dbUser && $auth0User && !$hasActiveBase) {
+        // check if deleted user that no active base is in sync with auth0
```
Comment (pylipp) [inline library/lib/auth0.php:260]: do you mean "check if deleted user without active base is in sync with Auth0" ?

File: library/lib/auth0.php [hunk 2]
```diff
@@ -207,20 +207,35 @@ function isUserInSyncWithAuth0($userId)
...
         $validationResult['name'] = ($auth0User['name'] == $dbUser['naam']) ? 'true' : 'false';
         $validationResult['usergroup_id'] = ($auth0User['app_metadata']['usergroup_id'] == $dbUser['cms_usergroups_id'] || null == $dbUser['cms_usergroups_id']) ? 'true' : 'false';
         $validationResult['organisation_id'] = ($auth0User['app_metadata']['organisation_id'] == $dbUser['organisation_id'] || null == $dbUser['cms_organisation_id']) ? 'true' : 'false';
-        if ($dbUser['base_ids']) {
-            $validationResult['base_ids'] = ($auth0User['app_metadata']['base_ids'] == array_intersect($auth0User['app_metadata']['base_ids'], $dbUser['base_ids'])) ? 'true' : 'false';
+
+        if ($dbUser['active_base_ids']) {
+            // check if all active bases are in sync
+            $validationResult['base_ids'] = ($auth0User['app_metadata']['base_ids'] == array_intersect($auth0User['app_metadata']['base_ids'], $dbUser['active_base_ids'])) ? 'true' : 'false';
```
Comment (pylipp) [inline library/lib/auth0.php:246]: Does this work if the lists from Auth0 and the DB have same content but different order?

Comment (vahidbazzaz) [reply library/lib/auth0.php]: @pylipp That's make sense, I fixed the logic

**Review by HaGuesto — APPROVED** [review]

**Comment by pylipp** [comment]
@vahidbazzaz this PR breaks the `active_users` check, in [the GCloud logs](https://console.cloud.google.com/logs/query;query=protoPayload.taskName%3D%2298111148451241425561%22%20protoPayload.taskQueueName%3D%22__cron%22;cursorTimestamp=2024-09-09T00:10:07.475604Z;startTime=2024-09-08T11:49:33.777Z;endTime=2024-09-09T08:49:33.777219Z?project=dropapp-242214) I see messages like 
```
PHP error: User with id 2 is out of sync between DB and Auth0 at base_ids in /workspace/library/lib/auth0.php on line 280
```
for most users in staging (all up to ID 39).

**Pull request merged**